lib/package.js

```diff
@@ -129,9 +79,9 @@ module.exports = function pkg(datastar, models) {
    *
    * @returns {Object} ?
    */
-  Pkg.fromPublish = function (payload) {
-    var replica = assign({}, this.extractLatest(payload));
-    var schemaFields = Pkg.schema.mappedFields();
+  function fromPublish({ pkg, payload }) {
+    var replica = assign({}, pkg.extractLatest(payload));
+    var schemaFields = model.schema.mappedFields();
```

Comment:
Since we can define schema as a separate variable in the scope of the model defining here, we can use it in the function here
```suggestion
    var schemaFields = Object.keys(schema);
```